src/task1/idsignificantchanges/idsignificantchanges.py

```diff
@@ -0,0 +1,180 @@
+    global second, canvas, point
+    """
+    cancels the creation of a point by removing the red point
+    """
+    canvas.delete(point)
+    second.destroy()
+
+
+def clicked(event):
+    """
+    a click on the map triggers this function.
+
+    -checks if coordinates of click fit inside the image dimensions, to
+    ensure that we can calculate lat/long validly
+    -calculates lat, long based on relative position of click to the x
+    and y dimension.
+    -puts a red dot on map to mark labelled location
+    -pops up window to add description to lat, long value.
+    -If save and quit is clicked, saves the point.
+
+    """
+    global index, lat, long, second, t, point
+    if (event.x >= 0 and event.x <= photo.width() and event.y >= 0 and event.y <= photo.height()):
+        second = tkinter.Toplevel(width = 300)
+
+        x1, y1 = (event.x - 1), (event.y - 1)
+        x2, y2 = (event.x + 1), (event.y + 1)
+
+        #Assumed that image is oriented North
+        #(relative y of point in image * distance between latitude coordinates) + most southern latitude coordinate
+        lat = (((photo.height()-event.y)/photo.height())*abs(inputmap['topleft']['lat']-inputmap['bottomright']['lat']))+inputmap['bottomright']['lat']
+        #(relative x of point in image * distance between longitude coordinates) + most western longitude coordinate
+        long = ((event.x/photo.width())*abs(inputmap['topleft']['long']-inputmap['bottomright']['long']))+inputmap['topleft']['long']
+        point = canvas.create_oval(x1, y1, x2, y2, fill="#FF0000", outline="#FF0000", width=10)
+        second.grab_set()
+        second.title("New Label")
+
+        #message box
+        msg = tkinter.Message(second, justify = 'center', width = 300, text="Label at (lat,long) \n{}, {}".format(lat, long))
+        msg.pack()
+        t = tkinter.Text(second, height=5, width=40)
+        t.pack()
+
+        button = tkinter.Button(second, text="Save and Quit", command=save)
+        button.pack()
+
+        second.protocol("WM_DELETE_WINDOW", delete)
+        second.mainloop()
+
+
+#Argparse section
+parser = argparse.ArgumentParser(description='You must specify paths to two jsons')
+parser.add_argument("--map", help = 'valid path to the reference map json', type = str)
+parser.add_argument("--photos", help = 'valid path to the survey photos json', type = str)
+args = parser.parse_args()
+mappath = args.map
+photopath = args.photos
+
+#read paths provided by command line arguments e.g.
+#python idsignificantchanges.py --map=inputmap.json --photos=inputsurveyphotos.json
+with open(mappath, 'r') as f:
+    inputmap = json.load(f)
+with open(photopath, 'r') as f:
+    inputsurveyphotos = json.load(f)
+
+#Create tk instance and set it to output the json on window close.
+top = tkinter.Tk()
+top.protocol("WM_DELETE_WINDOW", output)
+
+#Open the map image and fit the window to the image dims
+img = PIL.Image.open(inputmap['filename']);
+photo =PIL.ImageTk.PhotoImage(img)

```

Comment:
nit, but you might want to run this file through a linter :)

Reply:
done